drivers/adc_ng/include/adc_ng_internal.h

```diff
@@ -0,0 +1,267 @@
+ * production variations of +-100mV, but is extremely stable; so it will
+ * consistently create the same voltage for a wide range of supply voltage and
+ * operating temperatures. So if the actual value of the reference is e.g.
+ * written to EEPROM and restored on boot, the reference value should be marked
+ * as calibrated.
+ */
+#define ADC_NG_REF_CALIBRATED           (0x4000U)
+/**
+ * @brief   Access the voltage value without flags
+ */
+#define ADC_NG_REF_MASK                 (0x3fffU)
+
+/**
+ * @brief   Description of an thermistor to use for temperature measurements
+ */
+typedef struct {
+    /**
+     * @brief   Contains the temperature coefficient of the NTC, or zero
+     *
+     * The coefficient is is given in 1/1024 mV per 0.1 °C. The resulting
+     * temperature in 0.1 °C is calculated from the measured voltage using:
+     *
+     *    T[d°C] = (coefficient * mV) / 1024 + offset
+     */
+    uint16_t coefficient;
+    /**
+     * @brief   The offset (in 0.1 °C) to use for obtaining the temperature
+     *
+     * See @ref adc_ng_ntc_t::coefficient
+     */
+    int16_t offset;
+} adc_ng_ntc_t;
+
+/**
+ * @brief   Internal driver interface
+ */
+typedef struct {
+    /**
+     * @brief   Initialize the given ADC channel and prepare it for sampling
+     * @param           handle  Handle of the ADC
+     * @param[in]       chan    The ADC channel to initialize
+     * @param[in]       res     The resolution to select
+     * @param[in        ref     Index of the reference to use (@ref adc_ng_driver_t::refs)
+     *
+     * @retval  0               Success
+     * @retval  -ENXIO          Invalid channel given in @p channel
+     * @retval  -EALREADY       The ADC is already powered and configured
+     * @retval  <0              Other error (see device driver doc)
+     *
+     * @post    When 0 is returned, the channel @p channel is ready take samples
+     * @post    If @p res contains an unsupported resolution, an assert blows up
+     * @post    If `-EALREADY` is returned, the ADC is state remains unchanged
+     * @post    On other error codes the ADC is powered down
+     *
+     * @note    A call to @ref adc_ng_driver_t::off is needed to disable the channel
+     *          again and preserve power
+     */
+    int (*init)(void *handle, uint8_t chan, uint8_t res, uint8_t ref);
+    /**
+     * @brief   Disable the given ADC channel again and bring the ADC into a low
+     *          power state, unless other ADC channels are still onchannel
+     *
+     * @param           handle  Handle of the ADC
+     */
+    void (*off)(void *handle);
+    /**
+     * @brief   Runs a single conversion and returns the sample
+     *
+     * @param           handle  Handle of the ADC
+     * @param[out]      dest    The sample will be stored here
+     *
+     * @return  The result of the conversion
+     *
+     * @pre     The channel given in @p channel is currently initialized, see
```

Comment:
There is no parameter `channel`.

Reply:
This is related to the question concerning the channel parameter below.

Reply:
Fixed. Thanks!